internal/controllers/machine/machine_controller.go

```diff
@@ -816,14 +803,10 @@ func (r *Reconciler) drainNode(ctx context.Context, s *scope) (ctrl.Result, erro
 
 	remoteClient, err := r.ClusterCache.GetClient(ctx, util.ObjectKey(cluster))
 	if err != nil {
-		if errors.Is(err, clustercache.ErrClusterNotConnected) {
-			log.V(5).Info("Requeuing drain Node because connection to the workload cluster is down")
-			s.deletingReason = clusterv1.MachineDeletingDrainingNodeV1Beta2Reason
-			s.deletingMessage = "Requeuing drain Node because connection to the workload cluster is down"
-			return ctrl.Result{RequeueAfter: time.Minute}, nil
-		}
-		log.Error(err, "Error creating a remote client for cluster while draining Node, won't retry")
-		return ctrl.Result{}, nil
+		log.V(5).Info("Waiting for Cluster connection to come up to drain the Node")
+		s.deletingReason = clusterv1.MachineDeletingDrainingNodeV1Beta2Reason
+		s.deletingMessage = "Waiting for Cluster connection to come up to drain the Node"
```

Comment:
```suggestion
```
1. error shows up anyway because we propagate it up
2. deletingReason / deletingMessage are overwritten one level higher